### PR TITLE
adding information about the need of associations

### PR DESCRIPTION
**Pull request by wrongspot**
I did this tutorial and after part 1 I expected the extension to load automatically by product enity load. 
In my opinion, there is no information that additional associations are needed.

### CODE REVIEW

**Review by PaddyS — CHANGES_REQUESTED** [review]
Hey @wrongspot,

thank you for your pull request!

Unfortunately, I don't really understand why it wouldn't work without your addition.
Your extension should load automatically when you're using the parameter mentioned [here](https://github.com/shopware/docs/pull/374/files#diff-a067680a577ef19b4820d4138937d15d2e62a4e08486b05a6c95775b0cb76922L115).

I've quickly tested it my example plugin that I had back then and it definitely works for me. But maybe I am just missing something here, so I'd love to hear back from you! 🙏

File: guides/plugins/plugins/framework/data-handling/add-complex-data-to-existing-entities.md [hunk 1]
```diff
@@ -241,6 +241,65 @@ SQL;
...
+        ];
+    }
+
+    public function onProductCriteriaLoaded(ProductPageCriteriaEvent $event): void
+    {
+        $event->getCriteria()->addAssociation('exampleExtension');
+    }
+}
+```
+Of course, registering your subscriber via services.xml
```
[the remark this answers is not in the document]
Comment (wrongspot) [reply guides/plugins/plugins/framework/data-handling/add-complex-data-to-existing-entities.md:276]: Hi @PaddyS

My expectation from the original tutorial is that I will create an EntityExtension according to the example, and that this extension will be available after loading the Entity. There is no indication that I also have to adjust the criteria object. Without "->addAssociation ('exampleExtension') exampleExtension is never loaded. In my example I register a subscriber to modify criteria object for that needs.

[the remark this answers is not in the document]
Comment (wrongspot) [reply guides/plugins/plugins/framework/data-handling/add-complex-data-to-existing-entities.md:276]: Hey @PaddyS, sorry for late response. You are absolutely right, of course. I totally overlooked the 'autoload' flag and it was false in my case. With that the PR is of course obsolete.

Best Regards

[the remark this answers is not in the document]
Comment (PaddyS) [reply guides/plugins/plugins/framework/data-handling/add-complex-data-to-existing-entities.md:276]: Thank you for your feedback, appreciate the effort though! 💙

**Pull request closed without merging**